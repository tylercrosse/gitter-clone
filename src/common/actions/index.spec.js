import shortid      from 'shortid';
import * as actions from './index';

describe('actions', () => {
  it('addMessage should create ADD_MESSAGE action', () => {
    const message = {
      userId: '1231231',
      text: 'Use Redux',
      rawMarkup: '<p>Use Redux</p>',
      convo: 'react'
    };
    expect(actions.addMessage(message))
      .toMatchSnapshot();
  });

  it('fetchMessages should create [CALL_API] action', () => {
    const convo = 'chat';
    expect(actions.fetchMessages(convo))
    .toMatchSnapshot();
  });

  it('addConvo should create ADD_CONVO action', () => {
    const convo = {name: 'chat'};
    expect(actions.addConvo(convo))
      .toMatchSnapshot();
  });

  it('fetchConvos should create [CALL_API] action', () => {
    expect(actions.fetchConvos())
      .toMatchSnapshot();
  });

  it('signIn should create [CALL_API] action', () => {
    expect(actions.signIn({username: 'Dan'}))
      .toMatchSnapshot();
  });

  it('openCreateRoomModal should create OPEN_CREATE_ROOM_MODAL action', () => {
    expect(actions.openCreateRoomModal())
      .toEqual({
        type: 'OPEN_CREATE_ROOM_MODAL',
      });
  });

  it('closeModal should create CLOSE_MODAL action', () => {
    expect(actions.closeModal())
      .toEqual({
        type: 'CLOSE_MODAL',
      });
  });
});
