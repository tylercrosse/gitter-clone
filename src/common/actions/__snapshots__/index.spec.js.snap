// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions addConvo should create ADD_CONVO action 1`] = `
Object {
  "name": "chat",
  "type": "server.ADD_CONVO",
}
`;

exports[`actions addMessage should create ADD_MESSAGE action 1`] = `
Object {
  "convo": "react",
  "rawMarkup": "<p>Use Redux</p>",
  "text": "Use Redux",
  "type": "server.ADD_MESSAGE",
  "userId": "1231231",
}
`;

exports[`actions fetchConvos should create [CALL_API] action 1`] = `
Object {
  Symbol(Call API): Object {
    "endpoint": "null/api/convos",
    "method": "GET",
    "types": Array [
      "CONVOS_REQUEST",
      Object {
        "payload": [Function],
        "type": "CONVOS_SUCCESS",
      },
      "CONVOS_FAILURE",
    ],
  },
}
`;

exports[`actions fetchMessages should create [CALL_API] action 1`] = `
Object {
  Symbol(Call API): Object {
    "endpoint": "null/api/messages/chat",
    "method": "GET",
    "types": Array [
      "MESSAGES_REQUEST",
      Object {
        "payload": [Function],
        "type": "MESSAGES_SUCCESS",
      },
      "MESSAGES_FAILURE",
    ],
  },
}
`;

exports[`actions signIn should create [CALL_API] action 1`] = `
Object {
  Symbol(Call API): Object {
    "body": "{\\"username\\":{\\"username\\":\\"Dan\\"}}",
    "endpoint": "null/api/signIn",
    "headers": Object {
      "Content-Type": "application/json",
    },
    "method": "POST",
    "types": Array [
      "SIGN_IN_REQUEST",
      Object {
        "payload": [Function],
        "type": "SIGN_IN_SUCCESS",
      },
      "SIGN_IN_FAILURE",
    ],
  },
}
`;
